### PR TITLE
Add example for package extensions

### PR DESCRIPTION
**Pull request by schnellecom**
While experimenting with the new extensions system for packages I found out how it works from the documentation. 
I think it would be very helpful to have a minimal example as well, so I made what is working for me general and added it to the documentation.

### CODE REVIEW

File: doc/ref/gappkg.xml [hunk 1]
```diff
@@ -1299,6 +1299,23 @@ each being a record with the following components.
...
+As an example the following as part of the <F>PackageInfo.g</F> load the file <F>fileForB.gd</F> as soon as package <C>B</C> is loaded and <F>fileForC.gi</F> once package <C>C</> is loaded. 
+
+<Log><![CDATA[
+Extensions := [
+  rec(
+    needed := [ ["B", "0.6"] ],
+    filename := "./gap/fileForB.gd",
+  ),
+  rec(
+    needed := [ ["C", "0.6"] ],
```
Comment (fingolfin) [inline doc/ref/gappkg.xml:1313]: Maybe use a different version here compared to `B`?
```suggestion
    needed := [ ["C", "1.2"] ],
```

File: doc/ref/gappkg.xml [hunk 2]
```diff
@@ -1299,6 +1299,23 @@ each being a record with the following components.
 </Item>
 </List>
 
+<P/>
+
+As an example the following as part of the <F>PackageInfo.g</F> load the file <F>fileForB.gd</F> as soon as package <C>B</C> is loaded and <F>fileForC.gi</F> once package <C>C</> is loaded.
```
Comment (fingolfin) [inline doc/ref/gappkg.xml:1304]:
```suggestion
As an example suppose the following is part of the <F>PackageInfo.g</F>.
Then &GAP; will load the file <F>fileForB.gd</F> as soon as package
<C>B</C> is loaded in version 0.6 or newer, and <F>fileForC.gi</F> once
package <C>C</> is loaded in version 1.2 or newer. 
```

**Comment by fingolfin** [comment]
And you can apply multiple at once by "batching" them

**Comment by schnellecom** [comment]
Good hint, especially the batching. I used the web view but did not see that button. Will try it next time I have a suggestion

**Review by fingolfin — APPROVED** [review]
Thanks!

**Pull request merged**